bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,377 @@
+            )
+
+            # Sort the winner messages
+            winners = []
+            for i, winner in enumerate(winner_messages):
+                winners.append(winner)
+                if len(winner_messages) > i + 1:
+                    if winner_messages[i + 1][1]['score'] != winner[1]['score']:
+                        break
+                elif len(winner_messages) == (i + 1) + 1:  # The next element is the last element
+                    if winner_messages[i + 1][1]['score'] != winner[1]['score']:
+                        break
+
+            # one iteration is complete
+            await channel.send("Today's Spook Name Rate Game ends now, and the winner(s) is(are)...")
+
+            async with channel.typing():
+                await asyncio.sleep(1)  # give the drum roll feel
+
+                if winners:  # make sure there is a winner
+
+                    if len(winners) > 1:
+                        await channel.send(" and ".join(self.ping(win[1]['author']) for win in winners) + " !")
+                        score = winners[0][1]['score']
+                        # display winners, their names and scores
+                        await channel.send(f"Congratulations to all! You have a score of {score}!")
+                        names = ', '.join([f'{win[1]["name"]} ({self.ping(win[1]["author"])})' for win in winners])
+                        await channel.send(f"Your names were: {names}!")
+
+                    else:  # there is only one winner
+                        # There cant be 0 winners since we are checking that above
+
+                        winner = winners[0][1]
+                        await channel.send(f"{self.ping(winner['author'])}!")
+                        await channel.send(f"Congratulations {self.ping(winner['author'])}! You have a score of "
+                                           f"{winner['score']}!")
+                        await channel.send(f"Your name was: **{winner['name']}**.")  # display winner, name and score
+
+                    # Send random party emojis
+                    party = [random.choice([':partying_face:', ':tada:']) for _ in range(random.randint(1, 10))]
+                    await channel.send(" ".join(party))
+
+                else:
+                    # no participants in current iteration
+                    await channel.send('Hmm... Looks like no one participated! :cry:')
+
+            async with self.checking_messages:  # Acquire the lock to delete the messages
+                await self.messages.clear()  # reset the messages
+
+        # send the next name
+        self.name = f"{random.choice(self.first_names)} {random.choice(self.last_names)}"
+        await self.data.set('name', self.name)
+
+        await channel.send("Let's move on to the next name!\nAnd the next name is...\n"
+                           f"**{self.name}**!\nTry to spookify that... :smirk:")
```

Comment:
These lines formatting is not very good.

Reply:
Is it better now?